### PR TITLE
release-controller: branch for 4.7

### PR DESCRIPTION
**Pull request by droslean**
generated with:
```sh
#!/bin/bash

set -o errexit
set -o nounset
set -o pipefail

export conf='core-services/release-controller/images-ocp-4.6.yaml'
sed -e 's/4.6/4.7/g' "${conf}" > "${conf/4.6/4.7}"

export conf='core-services/release-controller/rpms-ocp-4.6.yaml'
sed -e 's/4-6/4-7/g' -e 's/4.6/4.7/g' "${conf}" > "${conf/4.6/4.7}"
cp "${conf/4.6/4.7}" "clusters/build-clusters/01_cluster/openshift/release-controller/$( basename "${conf/4.6/4.7}" )"

for repo in core-services/release-controller/_repos/*4.6*; do
	sed -e 's/4.6/4.7/g' "${repo}" > "${repo/4.6/4.7}"
done

sed -i 's/base-4-6/base-4-7/g' ci-operator/config/openshift/images/openshift-images-release-4.7.yaml

for release in core-services/release-controller/_releases/release-ocp-4.6*; do
	sed -e 's/4.6/4.7/g' "${release}" > "${release/4.6/4.7}"
done

export conf="ci-operator/jobs/openshift/release/openshift-release-release-4.6-periodics.yaml"
sed -e 's/4.6/4.7/g' -e 's/4.5/4.6/g' "${conf}" > "${conf/4.6/4.7}"

git add core-services/ clusters/ ci-operator/
git commit --signoff -m "release-controller: branch for 4.7"
```

follow-up manual step
```sh
#!/bin/bash

set -o errexit
set -o nounset
set -o pipefail

current_release="4.6"
future_release="4.7"

for arch in '' '-s390x' '-ppc64le'; do
	oc tag "rhcos/machine-os-content:${current_release}${arch}" "rhcos/machine-os-content:${future_release}${arch}" --as system:admin
	name="$( oc get istag -n rhcos "machine-os-content:${current_release}${arch}" -o jsonpath={.image.metadata.name} )"
	oc tag "quay.io/openshift-release-dev/ocp-v4.0-art-dev@${name}" "ocp${arch}/${future_release}-art-latest${arch}:machine-os-content" --as system:admin
done
```

### CODE REVIEW

File: ci-operator/jobs/openshift/release/openshift-release-release-4.7-periodics.yaml [hunk 1]
```diff
@@ -0,0 +1,6690 @@
...
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        # prow doesn't allow init containers or a second container
+        export PATH=$PATH:/tmp/bin
+        mkdir /tmp/bin
+        curl https://mirror.openshift.com/pub/openshift-v4/clients/oc/4.6/linux/oc.tar.gz | tar xvzf - -C /tmp/bin/ oc
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.7-periodics.yaml:33]: something is wrong here `clients/oc/4.6/linux`

Comment (droslean) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.7-periodics.yaml:33]: hm... yea that link doesn't exist. Shall we use 4.5 ?

**Comment by stevekuznetsov** [comment]
@droslean what is the status here? Did we end up needing to change the script?

**Comment by droslean** [comment]
> what is the status here? Did we end up needing to change the script?

No. I used the same script and replace only the versions. The script generates exactly the changes you see in this PR.

**Pull request merged**